hadoop-hdfs-project/hadoop-hdfs/src/main/java/org/apache/hadoop/hdfs/server/datanode/fsdataset/impl/ReplicaMap.java

```diff
@@ -120,15 +118,13 @@ ReplicaInfo get(String bpid, long blockId) {
   ReplicaInfo add(String bpid, ReplicaInfo replicaInfo) {
     checkBlockPool(bpid);
     checkBlock(replicaInfo);
-    try (AutoCloseDataSetLock l = lockManager.readLock(LockLevel.BLOCK_POOl, bpid)) {
-      LightWeightResizableGSet<Block, ReplicaInfo> m = map.get(bpid);
-      if (m == null) {
-        // Add an entry for block pool if it does not exist already
-        map.putIfAbsent(bpid, new LightWeightResizableGSet<Block, ReplicaInfo>());
-        m = map.get(bpid);
-      }
-      return  m.put(replicaInfo);
+    LightWeightResizableGSet<Block, ReplicaInfo> m = map.get(bpid);
+    if (m == null) {
+      // Add an entry for block pool if it does not exist already
+      map.putIfAbsent(bpid, new LightWeightResizableGSet<Block, ReplicaInfo>());
+      m = map.get(bpid);
     }
+    return m.put(replicaInfo);
```

Comment:
It's not safe here. If there is somebody  changing the `map` after line125 but before line127, the `replicaInfo` may not be able to added to `map`.